### PR TITLE
feat: remove_CLO_addon_install_and_all_tests

### PR DESCRIPTION
**Pull request by StevenTobin**
## Description

The cluster-logging addon that we currently use does not support GCP and with the addition of the cluster-logging-operator install from observability operator we no longer need fleet manager to install the addon once that version of OO is out.

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has been created for changes required on the client side

### CODE REVIEW

File: templates/service-template.yml [hunk 1]
```diff
@@ -505,11 +505,6 @@ parameters:
   description: ID of the kas-fleetshard addon
   value: "kas-fleetshard-operator-qe"
 
-- name: CLUSTER_LOGGING_OPERATOR_ADDON_ID
```
Comment (machi1990) [inline templates/service-template.yml:508]: let's also remove https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/1d61b2c5b14c046861aadb7be41db2aaf1c38be9/templates/service-template.yml#L1103

File: internal/kafka/internal/clusters/provider.go [hunk 2]
```diff
@@ -33,8 +33,6 @@ type Provider interface {
 	// Install the strimzi operator in a given cluster
 	InstallStrimzi(clusterSpec *types.ClusterSpec) (bool, error)
 	// Install the cluster logging operator for a given cluster
```
Comment (machi1990) [inline internal/kafka/internal/clusters/provider.go:35]:
```suggestion
```

File: internal/kafka/internal/clusters/provider.go [hunk 3]
```diff
@@ -33,8 +33,6 @@ type Provider interface {
...
-	InstallClusterLogging(clusterSpec *types.ClusterSpec, params []types.Parameter) (bool, error)
-	// Install the cluster logging operator for a given cluster
```
Comment (machi1990) [inline internal/kafka/internal/clusters/provider.go:37]: This comment should stay and it should read
```
	// Install the kas-fleetshard operator for a given cluster
``

File: internal/kafka/internal/clusters/provider.go [hunk 4]
```diff
@@ -33,8 +33,6 @@ type Provider interface {
 	// Install the strimzi operator in a given cluster
 	InstallStrimzi(clusterSpec *types.ClusterSpec) (bool, error)
 	// Install the cluster logging operator for a given cluster
-	InstallClusterLogging(clusterSpec *types.ClusterSpec, params []types.Parameter) (bool, error)
```
Comment (machi1990) [inline internal/kafka/internal/clusters/provider.go:36]: let's also remove https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/1d61b2c5b14c046861aadb7be41db2aaf1c38be9/internal/kafka/internal/clusters/standalone_provider.go#L284

File: templates/service-template.yml (hunk 1 above)
Comment (machi1990) [reply templates/service-template.yml:508]: Same as removing an entry for `CLUSTER_LOGGING_OPERATOR_ADDON_ID` from https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/1d61b2c5b14c046861aadb7be41db2aaf1c38be9/docs/deploying-kas-fleet-manager-to-openshift.md#deploy-kas-fleet-manager

**Review by machi1990 — COMMENTED** [review]
thanks for the PR @StevenTobin 

I've left a couple of comments and also the change needs saas update in AppInterface

**Comment by StevenTobin** [comment]
@machi1990 thanks. I think I also need to regenerate the moqs? I looked at the documentation but didn't see anything could you direct me to how that is done?

**Comment by machi1990** [comment]
> @machi1990 thanks. I think I also need to regenerate the moqs? I looked at the documentation but didn't see anything could you direct me to how that is done?

Yes, correct. Can you run `make generate code/fix` that should do it.

**Comment by StevenTobin** [comment]
@machi1990 That command resulted in a huge amount of changes including in files I didn't touch, is that expected?

**Comment by machi1990** [comment]
> @machi1990 That command resulted in a huge amount of changes including in files I didn't touch, is that expected?

Can you share the files names that are changed? For your changes I expect only `xxx_moq.go` files to be changed.

But recent @pawelpaszki has seen a similar issue where some unrelated OpenAPI changes were generated. I'd like to see if it is the same issue.

**Comment by StevenTobin** [comment]
@machi1990 sure this is the list

```
	modified:   internal/connector/internal/api/admin/private/api_connector_clusters_admin.go
	modified:   internal/connector/internal/api/admin/private/api_connector_namespaces_admin.go
	modified:   internal/connector/internal/api/admin/private/api_connector_types.go
	modified:   internal/connector/internal/api/private/api_connector_clusters_agent.go
	modified:   internal/connector/internal/api/public/api_connector_clusters.go
	modified:   internal/connector/internal/api/public/api_connector_namespaces.go
	modified:   internal/connector/internal/api/public/api_connector_service.go
	modified:   internal/connector/internal/api/public/api_connector_types.go
	modified:   internal/connector/internal/api/public/api_connectors.go
	modified:   internal/kafka/internal/api/admin/private/api_default.go
	modified:   internal/kafka/internal/api/private/api_agent_clusters.go
	modified:   internal/kafka/internal/api/public/api_default.go
	modified:   internal/kafka/internal/api/public/api_errors.go
	modified:   internal/kafka/internal/api/public/api_security.go
	modified:   internal/kafka/internal/clusters/clusterbuilder_moq.go
	modified:   internal/kafka/internal/clusters/provider_factory_moq.go
	modified:   internal/kafka/internal/clusters/provider_moq.go
	modified:   internal/kafka/internal/services/cloud_providers_moq.go
	modified:   internal/kafka/internal/services/cluster_placement_strategy_moq.go
	modified:   internal/kafka/internal/services/clusterservice_moq.go
	modified:   internal/kafka/internal/services/data_plane_cluster_service_moq.go
	modified:   internal/kafka/internal/services/data_plane_kafka_service_moq.go
	modified:   internal/kafka/internal/services/kafka_instance_types_moq.go
	modified:   internal/kafka/internal/services/kafkaservice_moq.go
	modified:   internal/kafka/internal/services/kas_fleetshard_operator_addon_moq.go
	modified:   internal/kafka/internal/services/observatorium_service_moq.go
	modified:   internal/kafka/internal/services/quota_service_factory_moq.go
	modified:   internal/kafka/internal/services/quotaservice_moq.go
	modified:   pkg/auth/auth_agent_service_moq.go
	modified:   pkg/client/aws/client_moq.go
	modified:   pkg/client/aws/route53_moq.go
	modified:   pkg/client/keycloak/client_moq.go
	modified:   pkg/client/keycloak/gocloak_moq.go
	modified:   pkg/client/ocm/client_moq.go
	modified:   pkg/client/ocm/idgenerator_moq.go
	modified:   pkg/client/redhatsso/client_moq.go
	modified:   pkg/services/authorization/authorization_moq.go
	modified:   pkg/services/sso/internal_keycloak_service_moq.go
	modified:   pkg/services/sso/keycloakservice_moq.go
	modified:   pkg/services/sso/osd_keycloak_service_moq.go
	modified:   pkg/workers/worker_interface_moq.go
```

**Comment by pawelpaszki** [comment]
> Thanks @StevenTobin
> 
> That's a long list. I think it is the same behavior that @pawelpaszki was seeing.
> 
> Can you commit and push only the `internal/kafka/internal/clusters/provider_moq.go` and `internal/kafka/internal/services/clusterservice_moq.go` and we'll see how it goes?

that's correct Manyanda. I am seeing issues when running `make openapi/generate` as well. would be nice to create an investigation Jira to find out what the issue is

**Comment by dhirajsb** [comment]
There were no changes to the OpenAPI, how is `make generate` creating new api files?? What are the differences?

**Comment by miguelsorianod** [comment]
> @machi1990 That command resulted in a huge amount of changes including in files I didn't touch, is that expected?

cc @StevenTobin

That might be due to having a different `moq` version than the one that is defined in the Makefile. Probably due to having already a local outdated version installed in the local `bin` folder of the project.

Could you recursively delete the `bin` folder in the project root directory and re-run the makefile commands again?

Before re-running the makefile commands you can also delete all moq files by being in the project root directory and running:
```
find | grep -vi vendor | grep -i _moq | xargs rm
```

Then when the makefile commands are run again they will all be re-generated from zero.

**Comment by miguelsorianod** [comment]
Is there any update about this @StevenTobin ?

Thanks

**Comment by miguelsorianod** [comment]
Hi @StevenTobin is it possible to update this now?

Thanks

**Comment by StevenTobin** [comment]
Ah, this was unintentional, My fork had become disconnected from the repo and I deleted and recreated as part of a seperate PR I was opening. I'll open a new PR for this from the new fork.

**Pull request closed without merging**